sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -5,183 +5,109 @@
 ```ts
 
 // @public
-export interface Context {
-    deleteValue(key: symbol): Context;
-    getValue(key: symbol): unknown;
-    setValue(key: symbol, value: unknown): Context;
-}
-
-// @public
-const context_2: ContextAPI;
-export { context_2 as context }
+export function createTracingClient(options: TracingClientOptions): TracingClient;
 
 // @public
-export interface ContextAPI {
-    active(): Context;
+export interface Instrumenter {
+    createRequestHeaders(tracingContext?: TracingContext): Record<string, string>;
+    parseTraceparentHeader(traceparentHeader: string): TracingSpanContext | undefined;
+    startSpan(name: string, spanOptions: InstrumenterSpanOptions): {
+        span: TracingSpan;
+        tracingContext: TracingContext;
+    };
+    withContext<CallbackArgs extends unknown[], Callback extends (...args: CallbackArgs) => ReturnType<Callback>>(context: TracingContext, callback: Callback, ...callbackArgs: CallbackArgs): ReturnType<Callback>;
 }
 
 // @public
-export function createSpanFunction(args: CreateSpanFunctionArgs): <T extends {
-    tracingOptions?: OperationTracingOptions | undefined;
-}>(operationName: string, operationOptions?: T | undefined, startSpanOptions?: SpanOptions | undefined) => {
-    span: Span;
-    updatedOptions: T;
-};
-
-// @public
-export interface CreateSpanFunctionArgs {
-    namespace: string;
-    packagePrefix: string;
+export interface InstrumenterSpanOptions extends TracingSpanOptions {
+    packageName: string;
+    packageVersion?: string;
+    tracingContext?: TracingContext;
 }
 
 // @public
-export type Exception = ExceptionWithCode | ExceptionWithMessage | ExceptionWithName | string;
-
-// @public
-export interface ExceptionWithCode {
-    code: string | number;
-    message?: string;
-    name?: string;
-    stack?: string;
+export interface OperationTracingOptions {
+    tracingContext?: TracingContext;
 }
 
 // @public
-export interface ExceptionWithMessage {
-    code?: string | number;
-    message: string;
-    name?: string;
-    stack?: string;
-}
+export type SpanStatus = {
```

Comment:
@witemple-msft mentioned that we should avoid anonymous types where possible as they don't render well in documentation. Do we want to make these named types (like SpanStatusSuccess and SpanStatusError)?

Reply:
or just SpanSuccess and SpanError ?

Reply:
I ended up going with `SpanStatusSuccess` and `SpanStatusError` - it felt to me like SpanSuccess / SpanError represent the status of the span, _not_ the status of the operation.

But... not sure if what I have is actually better 😄 so I can be convinced otherwise - let me know!